src/test/java/uk/gov/pay/directdebit/payments/services/TransactionServiceTest.java

```diff
@@ -246,6 +248,36 @@ public void paymentPaidOutFor_shouldSetPaymentAsSucceeded_andRegisterAPaidOutEve
         assertThat(transaction.getState(), is(SUCCESS));
     }
 
+    @Test
+    public void paymentFailedFor_shouldSetPaymentAsFailed_andRegisterAPaymentFailedEvent() {
+
+        Transaction transaction = TransactionFixture
+                .aTransactionFixture()
+                .withState(PENDING_DIRECT_DEBIT_PAYMENT)
+                .toEntity();
+
+        service.paymentFailedFor(transaction);
+
+        verify(mockedTransactionDao).updateState(transaction.getId(), FAILED);
+        verify(mockedPaymentRequestEventService).registerPaymentFailedEventFor(transaction);
```

Comment:
I almost wonder if these tests might each benefit from a `verifyNoMoreInteraction(mockedPaymentRequestEventService)` line

Reply:
Will add in the next PR. Thanks :)